### DISCUSSION TITLE
Feature request: cancel_previous_builds but only cancel queued builds

### DISCUSSION THREAD

**Discussion by vg-swift** [codemagic-ci-cd/codemagic-docs#1951] [Ideas]
### Use case

```Text
We push quite often, and with current options we have no builds at all because they get cancelled all the time or we have too many builds queued.
```

### Idea

```Text
When auto-triggering builds on a branch, would like to keep the running builds, but replace any queued builds from that branch.
```

**Comment by mikhail-tokarev** [comment]
thanks for the request, this is interesting problem to solve. You can consider to increase number of parallel builds to decrease queue size or you can use scheduled builds to ensure you always have built last version from your branches.

**Comment by vg-swift** [comment]
We don't really need a build for every single commit, but we would still like to have recent builds throughout the day for QA to be able to check at anytime.
We already have plans to schedule builds for other stuff like running unit tests, but this won't completely solve our issue.

**Comment by vg-swift** [comment]
Another issue is that with Unity, if we cancel a build, the license won't be returned, so we're stuck with a lot of queued outdated builds

**Reply by mikhail-tokarev** [reply]
I see, this is a very good point

**Reply by shniqq** [reply]
I have the exact same use-case, so I second this request 🤝

**Comment by vg-swift** [comment]
Any update on this?

**Comment by mikhail-tokarev** [comment]
hey! first of all I want to apologize for the silence.

We have discussed this request internally and as result we are going to introduce some improvements how to better handle Unity licenses if build failed or was canceled. However, it's not exact the same as the original request to cancel only queued builds, but it should do a trick.

Unfortunately this month is already planned but we aim to release it in Q4. I will keep this ticket open to update you once we have it's ready.

Sorry again for the delay!

**Reply by vg-swift** [reply]
Sounds good! Although the possibility to auto-cancel queued builds would still be a nice bonus. That's the default behaviour on Unity Cloud Build, so we're just used to it.

**Reply by kcbuilder** [reply]
waiting for this ^_^

**Reply by shniqq** [reply]
Me too 👀  👍

**Comment by shniqq** [comment]
Any news on this?

**Comment by FlavioEngel** [comment]
Bump here, it's been years, still waiting for this feature.....
